### ISSUE TITLE
local state query UTxOByAddress isn't compatible with legacy output transactions

### ISSUE THREAD

**Issue by falcucci** [txpipe/pallas#385]
Running out and comparing `cardano-node` versions incompatibilities founded [here](https://github.com/input-output-hk/mithril/pull/1403#issuecomment-1908217428) during a mithril integration, we have found that actually the localstate query UTxOByAddress aren't working as expected due to a deserialization issue.

After a quick immersion in the transactions' output, the [cardano-ledger](https://github.com/IntersectMBO/cardano-ledger/blob/f85ec6f2e0a80101009187f79ff154ab33a82a21/eras/babbage/impl/cddl-files/babbage.cddl#L79) CDDL means that we aren't mapping the correct struct properly accordingly with the diagnostic.

the *legacy* output is missing at pallas side to bring this compatibility using latest versions available.

TLDR; the cardano-node version has nothing to do with the deserialization issue once it's clear that it could happen following the CDDL, but needs investigation of the legacy output transactions behavior.
